compiler/ccgtypes.nim

```diff
@@ -734,6 +734,16 @@ proc getTypeDescAux(m: BModule, origTyp: PType, check: var IntSet): Rope =
       let cppName = getTypeName(m, t, sig)
       var i = 0
       var chunkStart = 0
+
+      template addResultType(ty: untyped) =
```

Comment:
This deserves to be a regular proc IMO.

Reply:
The code isn't used at all outside of this proc, I believe a template hits the sweet spot between a proc and copy&paste.